nwb_conversion_tools/NWBConverter.py

```diff
@@ -0,0 +1,58 @@
+"""Authors: Cody Baker and Ben Dichter."""
+from copy import deepcopy
+from .utils import get_schema_from_hdmf_class, get_root_schema
+from pynwb import NWBHDF5IO, NWBFile
+from pynwb.file import Subject
+from datetime import datetime
+import uuid
+
+
+class NWBConverter:
+    data_interface_classes = None  # dict of all data interfaces
+
+    @classmethod
+    def get_input_schema(cls):
+        """Compile input schemas from each of the data interface classes."""
+        input_schema = deepcopy(get_root_schema())
+        for name, data_interface in cls.data_interface_classes.items():
+            input_schema['properties'].update(data_interface.get_input_schema())
+        return input_schema
+
+    def __init__(self, **input_paths):
+        """Initialize all of the underlying data interfaces."""
+        self.data_interface_objects = {name: data_interface(**input_paths[name])
+                                       for name, data_interface in self.data_interface_classes.items()}
+
+    def get_metadata_schema(self):
+        """Compile metadata schemas from each of the data interface objects."""
+        metadata_schema = deepcopy(get_root_schema())
+        metadata_schema['properties'] = dict(
+            NWBFile=get_schema_from_hdmf_class(NWBFile),
+            Subject=get_schema_from_hdmf_class(Subject)
+        )
+        for name, data_interface in self.data_interface_objects.items():
+            metadata_schema['properties'].update(data_interface.get_metadata_schema()['properties'])
```

Comment:
```suggestion
            metadata_schema['properties'].update(name=data_interface.get_metadata_schema()['properties'])
```

Reply:
Will review and modify on the separate dictionary update PR.